### PR TITLE
Add libest/3.2.0

### PR DESCRIPTION
**Pull request by prince-chrismc**
Specify library name and version:  **libest/3.2.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Currently only Linux is supported, will look into windows in another PR. Not sure what todo about the gradle build system

### CODE REVIEW

**Comment by prince-chrismc** [comment]
:thinking: We do `tools.rmdir` all the time... but how do I zap a single file?

File: recipes/libest/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    def build(self):
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+        self.copy("*LICENSE", src=self._source_subfolder, dst="licenses")
+        with tools.chdir(self._source_subfolder):
+            autotools = self._configure_autotools()
+            autotools.install()
```
Comment (madebr) [inline recipes/libest/all/conanfile.py:69]:
```suggestion
        autotools.install()

    os.unlink(os.path.join(self.package_folder, "lib", "libest.la"))
```

File: recipes/libest/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.system_libs = ["dl", "pthread"]
```
Comment (madebr) [inline recipes/libest/all/conanfile.py:73]:
```suggestion
        self.cpp_info.libs = ["est"]
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["dl", "pthread"]
```

File: recipes/libest/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+
+class LibEstConan(ConanFile):
+    name = "libest"
+    license = "BSD-3"
```
Comment (madebr) [inline recipes/libest/all/conanfile.py:8]:
```suggestion
    license = "BSD-3-Clause"
```

File: recipes/libest/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    def _configure_autotools(self):
+        if not self._autotools:
+            self._autotools = AutoToolsBuildEnvironment(self)
+            # TODO: 
+            # - Static only build: https://github.com/cisco/libest/blob/70824ddc09bee661329b9416082d88566efefb32/intro.txt#L140
+            # - Release build: https://github.com/cisco/libest/blob/70824ddc09bee661329b9416082d88566efefb32/intro.txt#L253
+            args = []
+            # if not self.options.shared:
+            #     args.append("--disable-shared")
```
Comment (madebr) [inline recipes/libest/all/conanfile.py:56]:
```suggestion
            args = []
            if self.options.shared:
                 args.extend(["--enable-shared", "--disable-static"])
             else:
                 args.extend(["--disable-shared", "--enable-static"])
```

**Review by madebr — COMMENTED** [review]
libest is using libtool, so the `.la` file is created and installed always.
So you can just do `os.unlink`.
No need to test for existence or catching a `FileNotFoundError`.

File: recipes/libest/all/conanfile.py (hunk 1 above)
Comment (prince-chrismc) [inline recipes/libest/all/conanfile.py:69]:
```suggestion
        autotools.install()
        os.unlink(os.path.join(self.package_folder, "lib", "libest.la"))
```

Comment (prince-chrismc) [reply recipes/libest/all/conanfile.py]: Applied with corrected whitespace!

File: recipes/libest/all/conanfile.py (hunk 4 above)
Comment (prince-chrismc) [reply recipes/libest/all/conanfile.py]: Are these standard args? they were not documented

File: recipes/libest/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        if not self._autotools:
+            self._autotools = AutoToolsBuildEnvironment(self)
+            # TODO: 
+            # - Static only build: https://github.com/cisco/libest/blob/70824ddc09bee661329b9416082d88566efefb32/intro.txt#L140
+            # - Release build: https://github.com/cisco/libest/blob/70824ddc09bee661329b9416082d88566efefb32/intro.txt#L253
+            args = []
+            if self.options.shared:
+                 args.extend(["--enable-shared", "--disable-static"])
+             else:
+                 args.extend(["--disable-shared", "--enable-static"])
```
Comment (prince-chrismc) [inline recipes/libest/all/conanfile.py:58]:
```suggestion
            if self.options.shared:
                 args.extend(["--enable-shared", "--disable-static"])
            else:
                 args.extend(["--disable-shared", "--enable-static"])
```

**Comment by prince-chrismc** [comment]
@madebr Thanks for your expertise on this one =)

**Comment by madebr** [comment]
@prince-chrismc 
Please consider https://github.com/prince-chrismc/conan-center-index/pull/6
It looks like the examples are broken.
The patch just does not build them.

File: recipes/libest/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os in ("Windows", "Macos"):
```
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/libest/all/conanfile.py:32]: No I did not have the chance. Internally I just included in a vcxproj so I know it works on windows but that's as far as I've gotten

My intention was to tackle it later

**Review by danimtb — APPROVED** [review]

File: recipes/libest/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            self._autotools = AutoToolsBuildEnvironment(self)
+            # TODO: 
+            # - Static only build: https://github.com/cisco/libest/blob/70824ddc09bee661329b9416082d88566efefb32/intro.txt#L140
+            # - Release build: https://github.com/cisco/libest/blob/70824ddc09bee661329b9416082d88566efefb32/intro.txt#L253
+            args = []
+            if self.options.shared:
+                 args.extend(["--enable-shared", "--disable-static"])
+            else:
+                 args.extend(["--disable-shared", "--enable-static"])
+            self._autotools.configure(args=args)
```
Comment (uilianries) [inline recipes/libest/all/conanfile.py:60]:
```suggestion
            self._autotools.configure(args=args, configure_dir=self._source_subfolder)
```

You don't need to use tools.chdir

Comment (prince-chrismc) [reply recipes/libest/all/conanfile.py:60]: I did, when I tried it there paths were off by one directory

Comment (uilianries) [reply recipes/libest/all/conanfile.py:60]: I see, well, thanks for trying.

**Pull request merged**